ui/server/src/main/scala/pl/touk/nussknacker/ui/component/ComponentService.scala

```diff
@@ -113,13 +131,19 @@ object DefaultComponentService {
   }
 }
 
-class DefaultComponentService private (config: Config,
-                              processingTypeDataProvider: ProcessingTypeDataProvider[ProcessingTypeData],
-                              fetchingProcessRepository: FetchingProcessRepository[Future],
-                              subprocessRepository: SubprocessRepository,
-                              categoryService: ConfigProcessCategoryService)(implicit ec: ExecutionContext) extends ComponentService {
+/**
+  * componentsIdMap - it's prepared map, where key is ComponentId computed from componentsUiConfig
+  * and value is original componentName (declared in ProcessConfigCreator and placed id db)
+ */
+class DefaultComponentService private(componentsIdMap: Map[ComponentId, String],
```

Comment:
Shouldn't it be map ComponentId => 
union: 
(processingType(s), componentType, componentName) for "normal" components
componentName for "basic" components?

Reply:
I've added some tests for case where we have sharedComponent / notSharedComponentId and after that I think it should be `Map [ComponentId, List [ComponentId]]`. Because in sharedComponentId we joint many of components to one, eg.
`fraud-source-someSource` and` streaming-source-someSource` are mapped to `someSource`. So .. right now I try to handle this situation.